plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -4813,6 +4836,34 @@ public void testExpireSnapshotsPartitionedTable()
         assertThat(updatedSnapshots.size()).isLessThan(initialSnapshots.size());
     }
 
+    @Test
+    public void testExpireSnapshotsOnSnapshot()
+    {
+        Session session = Session.builder(getSession())
+                .setCatalogSessionProperty(getSession().getCatalog().orElseThrow(), "allow_legacy_snapshot_syntax", "true")
+                .build();
+        String tableName = "test_expire_snapshots_on_snapshot_" + randomTableSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a) AS VALUES 11", 1);
+        long snapshotId = getCurrentSnapshotId(tableName);
+        assertUpdate("INSERT INTO " + tableName + " VALUES 22", 1);
+        assertThatThrownBy(() -> query(session, "ALTER TABLE \"%s@%d\" EXECUTE EXPIRE_SNAPSHOTS".formatted(tableName, snapshotId)))
+                .hasMessage("Cannot execute table procedure EXPIRE_SNAPSHOTS on old snapshot " + snapshotId);
+        assertThat(query("SELECT * FROM " + tableName))
+                .matches("VALUES 11, 22");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testExpireSnapshotsSystemTable()
+    {
+        assertThatThrownBy(() -> query("ALTER TABLE \"nation$files\" EXECUTE EXPIRE_SNAPSHOTS"))
+                .hasMessage("This connector does not support table procedures");
```

Comment:
I know it is not in the scope of this PR but I find this error message missleading for this particular case. 
This connector actually supports table procedures just not on the system tables.

Reply:
I agree.

Reply:
Adding `io.trino.connector.system.SystemTablesMetadata#getTableHandleForExecute` with a more meaningful exception message would be a solution.